mithril-stm/src/multi_sig.rs

```diff
@@ -200,22 +200,11 @@ impl VerificationKeyPoP {
     // If we are really looking for performance improvements, we can combine the
     // two final exponentiations (for verifying k1 and k2) into a single one.
     pub fn check(&self) -> Result<(), MultiSignatureError> {
-        use blst::{
-            blst_fp12, blst_fp12_finalverify, blst_p1_affine_generator, blst_p2_affine_generator,
-            BLST_ERROR,
-        };
-        let result = unsafe {
-            let g1_p = *blst_p1_affine_generator();
-            let mvk_p = std::mem::transmute::<&BlstVk, &blst_p2_affine>(&self.vk.0);
-            let ml_lhs = blst_fp12::miller_loop(mvk_p, &g1_p);
-
-            let mut k2_p = blst_p1_affine::default();
-            blst_p1_to_affine(&mut k2_p, &self.pop.k2);
-            let g2_p = *blst_p2_affine_generator();
-            let ml_rhs = blst_fp12::miller_loop(&g2_p, &k2_p);
-
-            blst_fp12_finalverify(&ml_lhs, &ml_rhs)
-        };
+        // use blst::{
+        //     blst_fp12, blst_fp12_finalverify, blst_p1_affine_generator, blst_p2_affine_generator,
+        //     BLST_ERROR,
+        // };
+        let result = verify_miller_loop(&self.vk.0, &self.pop.k2);
```

Comment:
rename this function. This is not verifying the miller loop, but the full pairing. Also, I suggest that the function signatures are, as much as possible, using mithril-stm types, rather than blst types: 
```suggestion
        let result = verify_pairing(&self.vk, &self.pop);
```